dydxprotocol/v4-chain
# Add initial simple script to generate release notes

### Changelist
Add a script that generates a markdown list of changes between two commits.

### Test Plan
Tried the script on some examples:

`GITHUB_TOKEN=<redacted> python3 main.py --old release/protocol/v1.x --new main --path protocol` gives:
- [DEC-2082] add + clarify comments ([#495](https://github.com/dydxprotocol/v4-chain/pull/495))
- [CLOB-947] Split TEST-USD liquidity tier with differing base_position_notional ([#634](https://github.com/dydxprotocol/v4-chain/pull/634))
- Update default slashing params ([#636](https://github.com/dydxprotocol/v4-chain/pull/636))
- Conditionally info log in PlaceOrder DeliverTx if collat check fails ([#635](https://github.com/dydxprotocol/v4-chain/pull/635))
- [CLOB-177] - long term e2e tests ([#457](https://github.com/dydxprotocol/v4-chain/pull/457))
- fix edge case where position is closed before deleveraging ([#632](https://github.com/dydxprotocol/v4-chain/pull/632))
- [refactor] Remove duplicate CLOB `PrepareCheckState` latency metric ([#631](https://github.com/dydxprotocol/v4-chain/pull/631))
- Test exchange - use public dummy http server ([#630](https://github.com/dydxprotocol/v4-chain/pull/630))
- Update `pregenesis.sh` script for testnet4 ([#628](https://github.com/dydxprotocol/v4-chain/pull/628))
- [CLOB-947] New liquidity tier for TEST-USD ([#626](https://github.com/dydxprotocol/v4-chain/pull/626))
- [DEC-2081] Add tests for app/process/proposed_operations.go ([#609](https://github.com/dydxprotocol/v4-chain/pull/609))

`GITHUB_TOKEN=<redacted> python3 main.py --old a814748dfb39dc70302becbca95c19606e3bab8e --new release/indexer/v0.4.x --path indexer` gives (this tests cherry-picks that don't have associated PR):
- Revert "[IND-412]: Correctly set order status for all order types (#577)" ([0dbc3ee](https://github.com/dydxprotocol/v4-chain/commit/0dbc3eeeedc47a60c8ff309c80d8531e087da12f))
- Revert "Fix FOK orders so status is always FILLED (#594)" ([d720ed8](https://github.com/dydxprotocol/v4-chain/commit/d720ed85f227197a0569beb37c0a78964f21ef07))


### Author/Reviewer Checklist
